WordPress/gutenberg
# BlockLockModal: restore focus on fallback toolbar button when original button is not rendered

## What?


Improve focus restoration after un-locking a block via the lock button in the block toolbar

Partially addresses #51447

## Why?


Currently, when unlocking a block via the block lock modal opened by pressing the "lock" toolbar button, the focus is not restored correctly, since the "lock icon" button gets removed from the toolbar. As described in #51447, this is not ideal, especially for assistive technology users.

## How?




Added logic to the `BlockLockToolbar` component, so that the component moves focus to the first focusable button in the block toolbar when the `BlockLockModal` is closed and the button would otherwise disappear

## Testing Instructions





Check that focus restoration works as expected:

1. Select a block
2. In the block toolbar, open the "more" dropdown menu
3. Select the "lock" menu item
4. Check all the options in the modal
5. Click apply
6. Select the now locked block
7. In the block toolbar, click the "lock icon" toolbar _button_
8. Uncheck all the options in the modal
9. Click apply, make sure that the focus is restored correctly to another button in the toolbar
10. Select another block, click the "lock icon" toolbar button
11. Follow the steps 1 to 7 again
12. This time, do not uncheck the options in the modal, but simply click "apply"
13. Make sure that the focus is restored on the same "lock icon" button used to open the modal


Check that the are no regressions on the other usage of "BlockLockModal" via the "more" dropdown:

- Select a block
- In the block toolbar, open the "more" dropdown menu
- Select the "lock" menu item
- Check all the options in the modal
- Click apply, make sure that the focus is restored correctly on the button
- Select the "unlock" menu item
- Uncheck all the options in the modal
- Click apply, make sure that the focus is restored correctly on the button
- Select the "lock" menu item
- Click "cancel" in the modal, make sure that the focus is restored correctly on the button

## Screenshots or screencast 


https://github.com/WordPress/gutenberg/assets/1083581/da4876ad-53fe-4659-9003-a5612d916c79
